### DISCUSSION TITLE
Stream detection

### DISCUSSION THREAD

**Discussion by QuentinPerez** [hyperium/tonic#1304] [Q&A]
👋🏻

Is there a way to determine that it’s a request on a stream in a tower middleware ?

**Comment by LucioFranco** [comment]
You would need to check the path then check the filedescriptor set to see what that RPC is gonna do otherwise no.

**Reply by QuentinPerez** [reply]
Thank you for the answer, this is what I did. Do you think that we could have a better integration via the codegen ?

**Reply by LucioFranco** [reply]
The issue is a lot of this is encoded at the type system level, but if you have an idea to do this you can open an issue with a proposal and we can chat there.
